src/main/java/com/codeborne/selenide/hookactions/HookActions.java

```diff
@@ -0,0 +1,116 @@
+package com.codeborne.selenide.hookactions;
+
+import org.openqa.selenium.WebElement;
+
+import java.util.Collection;
+import java.util.HashMap;
+
+public class HookActions {
+  private static ThreadLocal<HookActions> instance = new ThreadLocal<>();
+  private HashMap<String, HookAction> beforeActions = new HashMap<>();
```

Comment:
ConcurrentHashMap?

Reply:
I just checked this case with
```
HookActions.getInstance().addBeforeAction("testHook", new TestAction("thread1", true));
new Thread(() -> {
  HookActions.getInstance().addBeforeAction("testHook", new TestAction("thread2", true));
  HookActions.getInstance().beforePerform(null, "command");
}).start();
Thread.sleep(500);
HookActions.getInstance().beforePerform(null, "command");
```
everything was fine.
ActionHooks with the same names was added and executed correctly.
one time per thread.